### PR TITLE
Feature/43/location-service-cities

### LINKED ISSUE #43: (SP: 2) Implement stepper Cities location service
## Description

Blocked by **#42**
Extend location service with cities (according to selected country) for the [issue](https://github.com/WebUI-Project-Based-07-vk/Client/issues/67) by using 
[countrystatecity.in](https://countrystatecity.in/) API

### PR DESCRIPTION
**Pull request by SeredenkoDS**
As an extension of #42, which pulls a list of countries using CountryStateCity API, this one, #43, pulls a list of cities.
Example usage: http://localhost:8080/get-city-list/UA (UA can be changed into an ISO2 code of any other country you see at http://localhost:8080/get-country-list )

When reviewing, check if the URL like http://localhost:8080/get-city-list/UA works for you, and if it does, feel free to approve. Thanks!

### CODE REVIEW

**Review by MrAK-dev — APPROVED** [review]

**Review by grbvtsk — APPROVED** [review]

File: src/services/locationServiceCities.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+const { createError } = require('~/utils/errorsHelper')
+const { COUNTRYSTATECITY_API_ISSUE } = require('~/consts/errors')
+
+const locationServiceCities = {
+  getCities: async (api_key, iso2_code) => {
+    var headers = new Headers()
```
Comment (vlad-khrychov) [inline src/services/locationServiceCities.js]:
```suggestion
    const headers = new Headers()
```

File: src/services/locationServiceCities.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    headers.append('X-CSCAPI-KEY', api_key)
+
+    var requestOptions = {
```
Comment (vlad-khrychov) [inline src/services/locationServiceCities.js]:
```suggestion
    const requestOptions = {
```

File: src/services/locationServiceCountries.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+const locationServiceCountries = {
+  getCountries: async (api_key) => {
+    var headers = new Headers()
```
Comment (vlad-khrychov) [inline src/services/locationServiceCountries.js]:
```suggestion
    const headers = new Headers()
```

File: src/services/locationServiceCountries.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    headers.append('X-CSCAPI-KEY', api_key)
+
+    var requestOptions = {
```
Comment (vlad-khrychov) [inline src/services/locationServiceCountries.js]:
```suggestion
    const requestOptions = {
```

File: src/services/locationServiceCountries.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+      redirect: 'follow'
+    }
+
+    try {
+      const response = await fetch('https://api.countrystatecity.in/v1/countries', requestOptions)
+      const result = await response.text()
+      return JSON.parse(result)
+    } catch (error) {
+      //
+    }
```
Comment (vlad-khrychov) [inline src/services/locationServiceCountries.js]: Throw here the custom error just like in Countries

**Review by vlad-khrychov — CHANGES_REQUESTED** [review]

**Review by olehpona — APPROVED** [review]

File: src/services/locationServiceCities.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    const requestOptions = {
+      method: 'GET',
+      headers: headers,
+      redirect: 'follow'
+    }
+
+    try {
+      const response = await fetch(`https://api.countrystatecity.in/v1/countries/${iso2_code}/cities`, requestOptions)
+      const result = await response.text()
+      return JSON.parse(result)
```
Comment (vlad-khrychov) [inline src/services/locationServiceCities.js]: Can we use here 
```suggestion
      const result = await response.json()
      return result 
```

**Review by vlad-khrychov — CHANGES_REQUESTED** [review]
You forget to delete `var`-s.

**Review by vlad-khrychov — APPROVED** [review]

**Pull request merged**